pkg/collector/corechecks/system/disk/disk/disk_nix.go

```diff
@@ -4,149 +4,168 @@
+	data, err := io.ReadAll(file)
 	if err != nil {
-		return err
+		return "", err
 	}
-	sender.Commit()
-
-	return nil
+	return string(data), nil
 }
 
-func (c *Check) collectPartitionMetrics(sender sender.Sender) error {
-	partitions, err := diskPartitions(true)
+func (c *Check) fetchAllDeviceLabelsFromBlkidCache() error {
+	log.Debugf("Fetching all device labels from blkid cache")
+	rawOutput, err := BlkidCacheCommand(c.cfg.blkidCacheFile)
 	if err != nil {
 		return err
 	}
-
-	for _, partition := range partitions {
-		if c.excludeDisk(partition.Mountpoint, partition.Device, partition.Fstype) {
+	log.Debugf("blkid cache output: %s", rawOutput)
+	c.deviceLabels = make(map[string]string)
+	lines := strings.Split(strings.TrimSpace(string(rawOutput)), "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		log.Debugf("processing line: '%s'", line)
+		if line == "" {
 			continue
 		}
-
-		// Get disk metrics here to be able to exclude on total usage
-		usage, err := diskUsage(partition.Mountpoint)
+		var device Device
+		err := xml.Unmarshal([]byte(line), &device)
 		if err != nil {
-			log.Warnf("Unable to get disk metrics of %s mount point: %s", partition.Mountpoint, err)
-			continue
-		}
-
-		// Exclude disks with total disk size 0
-		if usage.Total == 0 {
+			log.Debugf("Failed to parse line %s because of %v - skipping the line (some labels might be missing)\n", line, err)
 			continue
 		}
 
-		tags := make([]string, 0, 2)
-
-		if c.cfg.tagByFilesystem {
-			tags = append(tags, partition.Fstype, fmt.Sprintf("filesystem:%s", partition.Fstype))
+		if device.Label != "" && device.Text != "" {
+			c.deviceLabels[device.Text] = device.Label
 		}
-		var deviceName string
-		if c.cfg.useMount {
-			deviceName = partition.Mountpoint
-		} else {
-			deviceName = partition.Device
-		}
-		tags = append(tags, fmt.Sprintf("device:%s", deviceName))
-		tags = append(tags, fmt.Sprintf("device_name:%s", filepath.Base(partition.Device)))
-
-		tags = c.applyDeviceTags(partition.Device, partition.Mountpoint, tags)
-
-		c.sendPartitionMetrics(sender, usage, tags)
 	}
-
 	return nil
 }
 
-func (c *Check) collectDiskMetrics(sender sender.Sender) error {
-	iomap, err := ioCounters()
+var BlkidCommand = func() (string, error) {
+	cmd := exec.Command("blkid")
+	output, err := cmd.Output()
 	if err != nil {
-		return err
+		return "", err
 	}
-	for deviceName, ioCounter := range iomap {
-
-		tags := []string{}
-		tags = append(tags, fmt.Sprintf("device:%s", deviceName))
-		tags = append(tags, fmt.Sprintf("device_name:%s", deviceName))
-
-		tags = c.applyDeviceTags(deviceName, "", tags)
+	return string(output), nil
+}
 
-		c.sendDiskMetrics(sender, ioCounter, tags)
+func (c *Check) fetchAllDeviceLabelsFromBlkid() error {
+	log.Debugf("Fetching all device labels from blkid")
+	rawOutput, err := BlkidCommand()
+	if err != nil {
+		return err
 	}
+	c.deviceLabels = make(map[string]string)
+	labelRegex := regexp.MustCompile(`LABEL="([^"]+)"`)
```

Comment:
We should move this regex to a constant 😄

Reply:
Hey @jose-manuel-almaza thanks for moving the regex pattern to a constant. What I was suggesting was to move the entire `regexp.MustCompile(`LABEL="([^"]+)"`)` to a constant. 

It is a common pattern in Go to have the regex compile at start time, that way we do not throw a panic at runtime. 

From the Go doc: https://pkg.go.dev/regexp#MustCompile

> MustCompile is like [Compile](https://pkg.go.dev/regexp#Compile) but panics if the expression cannot be parsed. It simplifies safe initialization of global variables holding compiled regular expressions.

Other instances in the Agent codebase:
https://github.com/DataDog/datadog-agent/blob/c23991318bdae9f2137171b7d8d44e7753d2e346/pkg/util/scrubber/scrubber.go#L71-L72

Reply:
Sorry for the misunderstanding. It makes sense. Fixed.